src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementApiVersionSet.cs

```diff
@@ -0,0 +1,141 @@
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(PsApiManagementApiVersionSet), ParameterSetName = new[] { ExpandedParameterSet, ByInputObjectParameterSet })]
+    public class SetAzureApiManagementApiVersionSet : AzureApiManagementCmdletBase
+    {
+        #region Parameter Set Names
+
+        private const string ExpandedParameterSet = "ExpandedParameter";
+        private const string ByInputObjectParameterSet = "ByInputObject";
+
+        #endregion
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ParameterSetName = ByInputObjectParameterSet,
+            ValueFromPipeline = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementApiVersionSet. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementApiVersionSet InputObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier for new API Version Set.")]
+        public String ApiVersionSetId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "The name of the ApiVersion Set. This parameter is optional.")]
+        public String Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Versioning Scheme to select for the Api Versioning Set. This parameter is optional.")]
+        public PsApiManagementVersioningScheme Scheme { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "The Header value which will contain the versioning information. " +
+            "If versioning Scheme HEADER is choosen, then this value must be specified.")]
+        public String HeaderName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "The Query value which will contain the versioning information. " +
+            "If versioning Scheme Query is choosen, then this value must be specified.")]
+        public String QueryName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ExpandedParameterSet,
```

Comment:
@solankisamir all of the parameters that the user can update on a version set should be in _both_ parameter sets, or else you are setting `null` for all properties when the user provides `InputObject`